### DISCUSSION TITLE
Unity Smart Contract interaction with Truffle and Ganache

### DISCUSSION THREAD

**Discussion by Amirchk** [ConsenSys-archive/truffle#5953] [General Q&A]
I am building a blockchain game using unity engine on my locally created blockchain Ganache. 
Previously, I was using Chainsafe gaming SDK for unity web3 interaction with contracts. After I have created and deployed smart contract on ganache. I faced a provider error when interacting with smart contract from unity engine.

Now I am finding a way to have built-in web3 truffle library for unity engine which will also be localhost.

I hope you guys understand what i am saying.

**Comment by FlaviuWeb3** [comment]
Hi @Amirchk could you please elaborate a bit more about the issue you're currently facing?

**Reply by cds-amal** [reply]
cc: @OnlyOneJMJQ

**Reply by Amirchk** [reply]
@FlaviuWeb3 
Basically, I am building an NFT Game in unity.

1-Created smart contract to get **user balance** in solidity and deployed it on ganache localhost using TRUFFLE.
2- I have used ChainSafe SDK(which is used in UNITY ENGINE to interact with Smart Contracts written on solidity) so, I want to get user balance to be shown in the game. 
3- I have set chain ="Localhost 7545", network= "testnet", contractAddress="address", ABI="SomeABIofContract" and call balance function of SDK in unity.

Then all work well but It gives an issue "Provider is invalid or not set" 
I think it's a host issue. because I am using a local blockchain but calling EVM.Call() function of CHAINSAFE unity package which is an API call. I think I have to use a localhost provider.

**Reply by Victormaa** [reply]
hmm.. that's cool. Mark - to see what will this go next 😄
